tests/test_package_manager/test_v0.py

```diff
@@ -0,0 +1,188 @@
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Test package manager base."""
+
+
+import json
+import logging
+from collections import OrderedDict
+from pathlib import Path
+from unittest import mock
+
+from aea.package_manager.v0 import PackageManagerV0
+from aea.test_tools.test_cases import BaseAEATestCase
+
+from tests.test_package_manager.test_base import (
+    DUMMY_PACKAGE_HASH,
+    DUMMY_PACKAGE_ID,
+    EXAMPLE_PACKAGE_HASH,
+    EXAMPLE_PACKAGE_ID,
+    PACKAGE_JSON_FILE,
+)
+
+
+class TestPackageManagerV0(BaseAEATestCase):
+    """Test aea package manager."""
+
+    use_packages_dir: bool = True
+    packages_json_file: Path
+
+    def setup(
+        self,
+    ) -> None:
+        """Setup test."""
+        packages_v1 = json.loads(PACKAGE_JSON_FILE.read_text(encoding="utf-8"))
+
+        self.packages_json_file = self.packages_dir_path / "packages.json"
+        self.packages_json_file.write_text(
+            json.dumps({**packages_v1["dev"], **packages_v1["third_party"]})
+        )
+
+    def test_initialization(
+        self,
+    ) -> None:
+        """Test object initialization."""
+
+        pm = PackageManagerV0.from_dir(self.packages_dir_path)
+
+        assert len(pm.packages) == len(json.loads(self.packages_json_file.read_text()))
+        assert pm.path == self.packages_dir_path
+        assert pm.get_package_hash(package_id=EXAMPLE_PACKAGE_ID) is not None
+        assert pm.get_package_hash(package_id=DUMMY_PACKAGE_ID) is None
+
+    def test_sync(
+        self,
+    ) -> None:
+        """Test sync."""
+
+        pm = PackageManagerV0(
+            path=self.packages_dir_path,
+            packages=OrderedDict({DUMMY_PACKAGE_ID: DUMMY_PACKAGE_HASH}),
+        )
+
+        with mock.patch.object(pm, "add_package") as update_patch:
+            pm.sync()
+            update_patch.assert_called_with(package_id=DUMMY_PACKAGE_ID)
+
+
+class TestHashUpdate(BaseAEATestCase):
+    """Test hash update."""
+
+    use_packages_dir: bool = True
+
+    def test_update_package_hashes(
+        self,
+    ) -> None:
+        """Test update package hash method."""
+
+        packages_v1 = json.loads(PACKAGE_JSON_FILE.read_text(encoding="utf-8"))
+        packages = {**packages_v1["dev"], **packages_v1["third_party"]}
+
+        original_hash = packages[EXAMPLE_PACKAGE_ID.to_uri_path]
+        packages[EXAMPLE_PACKAGE_ID.to_uri_path] = DUMMY_PACKAGE_HASH
+
+        packages_json_file = self.packages_dir_path / "packages.json"
+        packages_json_file.write_text(json.dumps(obj=packages))
+        pm = PackageManagerV0.from_dir(self.packages_dir_path)
```

Comment:
tests should never operate on anything other than copies of files when doing anything other than reading